Examples/Jax/flax_example.py

```diff
@@ -0,0 +1,46 @@
+import netket as nk
+import flax
+from flax import linen as nn
+import jax
+from jax import numpy as jnp
+import numpy as np
+
+
+# 1D Lattice
+L = 20
+g = nk.graph.Hypercube(length=L, n_dim=1, pbc=True)
+
+# Hilbert space of spins on the graph
+hi = nk.hilbert.Spin(s=0.5) ** L
+
+ha = nk.operator.Ising(h=1.0, hilbert=hi, graph=g)
+
+alpha = 1
+
+
+class RBM(nn.Module):
+    @nn.compact
+    def __call__(self, x):
+        x = nn.Dense(features=alpha * x.shape[-1], dtype=jax.numpy.float32)(x)
+        x = jnp.log(jnp.cosh(x))
+        return jnp.sum(x, axis=-1)
+
+
+ma = nk.machine.Flax(hi, RBM())
+ma.init_random_parameters(seed=1232)
+
+# Jax Sampler
+sa = nk.sampler.MetropolisLocal(machine=ma, n_chains=2)
+
+# Using Sgd
+op = nk.optimizer.Sgd(ma, learning_rate=0.1)
```

Comment:
are we keeping these or just using the new optimizers ? I think it's not too bad to leave some convenience ```nk.optimizers``` names that just export the flax optimizers...

Reply:
the argument I have against ```nk.optimizers``` (plural) is that all the others are singular (```nk.sampler``` not samplers etc) and ```optim``` would be the only short version we are using

Reply:
This is part of the old PR #490 . It's not really about this.

However: `nk.optim` re-exports `flax.optim`, achieving what you want.

Reply:
Istillo have to update the examples